### ISSUE TITLE
[ty] lint:unresolved-reference for globals

### ISSUE THREAD

**Issue by mitsuhiko** [astral-sh/ruff#17992]
### Summary

```python
_foo = False

def stuff():
    global _foo
    if not _foo:
        _foo = True
```

Gives the following type check warning:

```
warning: lint:unresolved-reference: Name `_foo` used when not defined
 --> test.py:5:12
  |
3 | def stuff():
4 |     global _foo
5 |     if not _foo:
  |            ^^^^
6 |         _foo = True
  |
info: `lint:unresolved-reference` is enabled by default
```

### Version

ty 0.0.0-alpha.7 (905a3e1e5 2025-05-07)

**Comment by ntBre** [comment]
I think this should be fixed by https://github.com/astral-sh/ruff/pull/17637, which should come out in alpha 8 momentarily!

https://github.com/astral-sh/ty/pull/298
